be/test/storage/sstable/persistent_index_sstable_test.cpp

```diff
@@ -0,0 +1,366 @@
+    // seek & check
+    sstable::Table* sstable;
+    ASSIGN_OR_ABORT(auto read_file, fs::new_random_access_file(lake::join_path(kTestDir, filename)));
+    CHECK_OK(sstable::Table::Open(options, read_file.get(), filesz, &sstable));
+    sstable::ReadOptions read_options;
+    sstable::Iterator* iter = sstable->NewIterator(read_options);
+    for (int i = 0; i < 100; i++) {
+        int r = rand() % N;
+        iter->Seek(fmt::format("test_key_{:016X}", r));
+        ASSERT_TRUE(iter->Valid() && iter->status().ok());
+        ASSERT_TRUE(iter->key().to_string() == fmt::format("test_key_{:016X}", r));
+        IndexValue exp_val(r);
+        IndexValue cur_val(UNALIGNED_LOAD64(iter->value().get_data()));
+        ASSERT_TRUE(exp_val == cur_val);
+    }
+}
+
+TEST_F(PersistentIndexSstableTest, test_merge) {
+    std::vector<sstable::Iterator*> list;
+    std::vector<std::unique_ptr<RandomAccessFile>> read_files;
+    std::vector<uint64_t> fileszs;
+    fileszs.resize(3);
+    read_files.resize(3);
+    const int N = 10000;
+    for (int i = 0; i < 3; ++i) {
+        sstable::Options options;
+        const std::string filename = fmt::format("test_merge_{}.sst", i);
+        ASSIGN_OR_ABORT(auto file, fs::new_writable_file(lake::join_path(kTestDir, filename)));
+        sstable::TableBuilder builder(options, file.get());
+        for (int j = 0; j < N; j++) {
+            std::string str = fmt::format("test_key_{:016X}", j);
+            IndexValue val(j * i);
+            builder.Add(Slice(str), Slice(val.v, 8));
+        }
+        CHECK_OK(builder.Finish());
+        uint64_t filesz = builder.FileSize();
+        fileszs[i] = filesz;
+        ASSIGN_OR_ABORT(read_files[i], fs::new_random_access_file(lake::join_path(kTestDir, filename)));
+    }
+    sstable::Options options;
+    sstable::ReadOptions read_options;
+    for (int i = 0; i < 3; ++i) {
+        sstable::Table* sstable;
+        CHECK_OK(sstable::Table::Open(options, read_files[i].get(), fileszs[i], &sstable));
+        sstable::Iterator* iter = sstable->NewIterator(read_options);
+        list.emplace_back(iter);
+    }
+
+    phmap::btree_map<std::string, std::string> map;
+    {
+        sstable::Options options;
+        sstable::Iterator* iter = sstable::NewMergingIterator(options.comparator, &list[0], list.size());
+
+        iter->SeekToFirst();
+        while (iter->Valid()) {
+            auto key = iter->key().to_string();
+            auto it = map.find(key);
+            if (it == map.end()) {
+                map[key] = iter->value().to_string();
+            } else {
+                auto val = UNALIGNED_LOAD64(it->second.c_str());
+                auto cur_val = UNALIGNED_LOAD64(iter->value().get_data());
+                if (cur_val > val) {
+                    it->second = iter->value().to_string();
+                }
+            }
+            iter->Next();
+        }
+    }
+    {
+        phmap::btree_map<std::string, std::string> map2;
```

Comment:
Why do we need map2?

Reply:
Because we will use map later for more check

Reply:
But map2 is not used now.

Reply:
Done